src/main/java/chronosacaria/mcda/registry/ArmorsRegistry.java

```diff
@@ -50,87 +50,32 @@ protected static void registerArmor(ArmorSets set, EnumSet<EquipmentSlot> slots)
-
-        registerArmor(ArmorSets.PHANTOM,    EnumSet.of(HEAD, CHEST, LEGS, FEET));
-        registerArmor(ArmorSets.FROST_BITE, EnumSet.of(HEAD, CHEST, LEGS, FEET));
-
-        registerArmor(ArmorSets.PLATE,      EnumSet.of(HEAD, CHEST, LEGS, FEET));
-        registerArmor(ArmorSets.FULL_METAL, EnumSet.of(HEAD, CHEST, LEGS, FEET));
-
-        registerArmor(ArmorSets.REINFORCED_MAIL, EnumSet.of(HEAD, CHEST, LEGS, FEET));
-        registerArmor(ArmorSets.STALWART_MAIL,   EnumSet.of(HEAD, CHEST, LEGS, FEET));
-
-        registerArmor(ArmorSets.SCALE_MAIL, EnumSet.of(CHEST, LEGS, FEET));
-        registerArmor(ArmorSets.HIGHLAND,   EnumSet.of(HEAD, CHEST, LEGS, FEET));
-
-        registerArmor(ArmorSets.SNOW,  EnumSet.of(HEAD, CHEST, LEGS, FEET));
-        registerArmor(ArmorSets.FROST, EnumSet.of(HEAD, CHEST, LEGS, FEET));
-
-        registerArmor(ArmorSets.SOUL_ROBE,  EnumSet.of(HEAD, CHEST, LEGS, FEET));
-        registerArmor(ArmorSets.SOULDANCER, EnumSet.of(HEAD, CHEST, LEGS, FEET));
-
-        registerArmor(ArmorSets.SPELUNKER,    EnumSet.of(HEAD, CHEST, LEGS, FEET));
-        registerArmor(ArmorSets.CAVE_CRAWLER, EnumSet.of(HEAD, CHEST, LEGS, FEET));
-
-        registerArmor(ArmorSets.THIEF,  EnumSet.of(HEAD, CHEST, LEGS, FEET));
-        registerArmor(ArmorSets.SPIDER, EnumSet.of(HEAD, CHEST, LEGS, FEET));
-
-        registerArmor(ArmorSets.WOLF,       EnumSet.of(HEAD, CHEST, LEGS, FEET));
-        registerArmor(ArmorSets.BLACK_WOLF, EnumSet.of(HEAD, CHEST, LEGS, FEET));
-
-        registerArmor(ArmorSets.FOX,        EnumSet.of(HEAD, CHEST, LEGS, FEET));
-        registerArmor(ArmorSets.ARCTIC_FOX, EnumSet.of(HEAD, CHEST, LEGS, FEET));
-
-        registerArmor(ArmorSets.EMERALD, EnumSet.of(HEAD, CHEST, LEGS, FEET));
-        registerArmor(ArmorSets.OPULENT, EnumSet.of(HEAD, CHEST, LEGS, FEET));
-        registerArmor(ArmorSets.GILDED,  EnumSet.of(HEAD, CHEST, LEGS, FEET));
-
-        registerArmor(ArmorSets.CLIMBING_GEAR,        EnumSet.of(HEAD, CHEST, LEGS, FEET));
-        registerArmor(ArmorSets.RUGGED_CLIMBING_GEAR, EnumSet.of(HEAD, CHEST, LEGS, FEET));
-        // registerArmor(ArmorSets.GOAT,                 EnumSet.of(HEAD, CHEST, LEGS, FEET)); // TODO
+        for (ArmorSets set : ArmorSets.values()) {
+            EnumSet<EquipmentSlot> slots;
+
+            switch (set) {
+                case BATTLE:
+                case SPLENDID:
+                    slots = EnumSet.of(CHEST, LEGS);
+                    break;
+                case EVOCATION:
+                case EMBER:
+                case VERDANT:
+                    slots = EnumSet.of(HEAD, CHEST, LEGS);
+                    break;
+                case SCALE_MAIL:
+                    slots = EnumSet.of(CHEST, LEGS, FEET);
+                    break;
+                case HUNTER:
+                    slots = EnumSet.of(CHEST);
+                    break;
+                case ROYAL: continue; // TODO: remove when implementing royal armor
```

Comment:
Once I finally have the textures for the Royal armour set, would I just add the 
`slots = EnumSet.of(HEAD, CHEST, LEGS, FEET);` and then break after that whilst removing the `continue`?

Reply:
It would be better to remove the whole `case ROYAL: continue;` line, because the `switch` has a `default` case which sets `slots = EnumSet.of(HEAD, CHEST, LEGS, FEET)` if it didn't go in any other branch.

You only need to add a `case` if you don't want to register all slots.

Reply:
Ohh okay. That makes sense. Thanks for explaining that for me. :)